src/pages/ProductsPage/Products.tsx

```diff
@@ -6,37 +6,103 @@ import ReactPaginate from "react-paginate";
 import Product from "../../components/ui/productCard/ProductCard";
 import Loader from "../../components/ui/loader/Loader";
 import { useFetchBreedsQuery } from "../../store/redux-paginstion-slice/fetch-slice";
-import { useAppDispatch } from "../../shared/utils";
+import { useAppDispatch, usePagination } from "../../shared/utils";
 import { changePage } from "../../store/redux-paginstion-slice/fetch-slice";
+import Icon from "../../components/ui/icons/icon";
+import { BreedType } from "../../types";
 
 export default function Products() {
+  const _ = require("lodash");
   const [search] = useSearchParams();
-  const currentPage = Number(search.get("page"));
+  const currentPage = 1 && Number(search.get("page"));
   const dispatch = useAppDispatch();
-  const [page, setPage] = useState(currentPage);
-  const { data = [] } = useFetchBreedsQuery(page - 1);
+  const [pageNum, setPageNum] = useState(currentPage);
+  const { data = [] } = useFetchBreedsQuery(pageNum - 1);
+  const [searchBreeds, setSearchBreeds] = useState<BreedType[]>([]);
   let breeds = data;
   const pageCount = 17;
   const history = useNavigate();
+  const [searchTerm, setSearchTerm] = useState(search.get("search"));
+  const { firstContentIndex, page, lastContentIndex, setPage, totalPages } =
+    usePagination({
+      numberOfCards: 10,
+      listLength: searchBreeds.length,
+    });
 
   useEffect(() => {
+    history(
+      searchTerm
+        ? `/products?search=${searchTerm}`
+        : `/products?page=${pageNum}`
+    );
+    dispatch(changePage(pageNum));
+
+    if (searchTerm) {
+      const fetching = _.debounce(() => {
+        fetch("https://api.thedogapi.com/v1/breeds/search?q=dog")
```

Comment:
to me there is something incorrect - why `search?q=dog`?

You make a request to server for all entities with dog in name and then filter, right?
You must make a request with your searchTerm like this - https://api.thedogapi.com/v1/breeds/search?q=${searchTerm} and not filter then